src/main/java/teammates/storage/api/CommentsDb.java

```diff
@@ -199,92 +126,40 @@ public CommentAttributes getComment(CommentAttributes commentToGet) {
         Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, commentViewerType);
         
         List<Comment> comments = getCommentEntitiesForCommentViewer(courseId, commentViewerType);
-        List<CommentAttributes> commentAttributesList = new ArrayList<CommentAttributes>();
-        
-        for (Comment comment : comments) {
-            commentAttributesList.add(new CommentAttributes(comment));
-        }
-        return commentAttributesList;
+        return getAttributesListFromEntitiesList(comments);
     }
     
-    /*
-     * Get comment for the sending state (SENT|SENDING|PENDING)
-     */
+    @SuppressWarnings("unchecked")
     public List<CommentAttributes> getCommentsForSendingState(String courseId, CommentSendingState state) {
-        Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, courseId);
-        
-        List<Comment> comments = getCommentEntitiesForSendingState(courseId, state);
-        List<CommentAttributes> commentAttributesList = new ArrayList<CommentAttributes>();
-        
-        for (Comment comment : comments) {
-            commentAttributesList.add(new CommentAttributes(comment));
-        }
-        return commentAttributesList;
+        return (List<CommentAttributes>) super.getCommentsForSendingState(courseId, "PLACEHOLDER", state);
     }
     
     /*
      * Get comments for a course
      */
+    @SuppressWarnings("unchecked")
     public List<CommentAttributes> getCommentsForCourse(String courseId) {
-        Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, courseId);
-        
-        List<Comment> comments = getCommentEntitiesForCourse(courseId);
-        List<CommentAttributes> commentAttributesList = new ArrayList<CommentAttributes>();
-        
-        for (Comment comment : comments) {
-            commentAttributesList.add(new CommentAttributes(comment));
-        }
-        return commentAttributesList;
+        return (List<CommentAttributes>) super.getCommentsForCourse(courseId);
```

Comment:
there're many this kind of methods, only call super & do the conversion.
can we 
1. just remove them
2. create one method to do the conversion for respective logic.